pshtt/pshtt.py

```diff
@@ -602,16 +599,24 @@ def https_check(endpoint):
         cert_plugin_result = scanner.run_scan_command(server_info, command)
     except Exception as err:
         try:
-            if("timed out" in str(err)):
-                logging.warning("{}: Retrying sslyze scanner certificate plugin.".format(endpoint.url))
+            if "timed out" in str(err):
+                logging.exception("{}: Retrying sslyze scanner certificate plugin.".format(endpoint.url))
                 cert_plugin_result = scanner.run_scan_command(server_info, command)
+            else:
+                logging.exception("{}: Unknown exception in sslyze scanner certificate plugin.".format(endpoint.url))
+                utils.debug("{}: {}".format(endpoint.url, err))
+                endpoint.unknown_error = True
+                # We could make this False, but there was an error so
+                # we don't know
+                endpoint.https_valid = None
```

Comment:
Instead of picking `True` or `False` when there's an unknown exception, I suggest having the scan error out and report no data for this domain. It doesn't need to crash the entire scanning process, but I think it would be healthier for error classes that the `pshtt` maintainers haven't yet encountered to create some user-visible and dev-visible friction that incentivizes tracking down the new exception class and handling it appropriately.

I also think that from a customer perspective, it's better for `pshtt` to go from Nothing to True/False than to go from True to True/False as it gets more precise over time.

Reply:
@konklone, Is setting `unknown_error` to `True` sufficient to error out, or is there more that is needed?

I completely agree that it's better to return Nothing if we don't know for sure rather than True which could then later turn into a False.